include/envoy/request_id_extension/request_id_extension.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
```

Comment:
I don't feel strongly about it but should we put this in include/envoy/http since it's HTTP specific? WDYT?